### ISSUE TITLE
Missing shared project if using csproj imports with properties in file name

### ISSUE THREAD

**Issue by shadow-cs** [stryker-mutator/stryker-net#317]
Stryker throws `System.IO.FileNotFoundException: Missing shared project $(BuildDirectory)Foo.targets` if some of the `Imports` have properties in them (it is likely those properties are set in some of the MSBuild explicit imports as part of more complex build).

## Steps to reproduce

Add an `Import` with a `Project` containing a property (like the following snippet) to the `csproj` file of the project under test.

```xml
<Import Project="$(BuildDirectory)Foo.targets" />
```

## Related issues

* #235 
* #230

**Comment by shadow-cs** [comment]
I do understand that parsing the MSBuild XMLs including property evaluation is rather difficult. Sufficient solution might be adding an exclusion pattern that would skip certain `Import`s.

Another MSBuild features (like `Conditions`) may possibly break Stryker parsing as well.

**Comment by Mobrockers** [comment]
We are working in integrating [buildalyzer](https://buildalyzer.netlify.com/) which should be able to parse all msbuild config for us automatically. Once we have that it will be very easy to add these sort of misc project types and properties.

**Comment by richardwerkman** [comment]
Indeed, I hope to finish the buildalyzer integration quickly. I'll make sure that the shared projects are also read using buildalyzer. We will keep this issue open to keep this in mind and validate that this works once buildalyzer integration is done.

**Comment by richardwerkman** [comment]
Bad news... It seems that Buildalyzer doesn't resolve imports like we expected. But it gives us a list of available properties that we can replace with placeholders found in imports. We can look for `$()` inside imports and look for the contents in the list of properties buildalyzer gives us. The list contains over 500 msbuild properties. It contains for example:
```
{[MSBuildProjectDirectory, C:\Dev\ExampleProjects\TestApp\TestApp]}
```
If the import looked like:
``` xml
<Import Project="$(MSBuildProjectDirectory)\sharedproject1.\SharedProject1.projitems">
```
it would work I guess. 
Would this solve your issue @shadow-cs ?

**Comment by shadow-cs** [comment]
The trick is I'm using `Directory.Build.props` that initializes some of the properties. If Buildalyzer  does pick up these explicit imports then it would work I guess.

I would be happy to provide you with some test results. Would it help if I ran Buildalyzer on my project? I guess I can create some minimal setup that you can use to test my scenario if you wish.

**Comment by richardwerkman** [comment]
If you could test by running this [PR ](https://github.com/stryker-mutator/stryker-net/pull/318)on your project and validate that the output contains your properties that would be great! The output can be found [here](https://github.com/stryker-mutator/stryker-net/blob/7087e16934c04b35991d3eae9ed2b1996acdd534/src/Stryker.Core/Stryker.Core/Initialisation/InputFileResolver.cs#L61). It should be in `result.ProjectUnderTestAnalyzerResult` in the `Properties` list.

**Comment by shadow-cs** [comment]
Yes the `result.ProjectUnderTestAnalyzerResult.Properties` does include properties set by my `Directory.Build.props` with correct values.

**Comment by richardwerkman** [comment]
@shadow-cs Great! After #318 has been merged we can create a fix for this issue then. Would you have time to pick this up?

**Comment by shadow-cs** [comment]
@richardwerkman I think I might ;-)

**Comment by shadow-cs** [comment]
@richardwerkman I started working on this, the approach can be tracked in the commit I just pushed. I wanted to add a few integration tests, is there some guide how to do this (or am I free to just modify or add any projects I see fit)? Should I start a MR to the `full-framework` branch and push the changes there that will be squashed later?

**Comment by Mobrockers** [comment]
Keep in mind that currently the integration tests are ran for very single
build, so any project you add will add extra time to the build. It might be
better to create unit tests wherein you use an actual buildalyzer build but
not a complete Stryker run to validate. This should not add so much time.

**Comment by richardwerkman** [comment]
Yes I agree. This could also be achieved with a unit test which saves time! We have a wrapper around buildalyzer so you could mock the msbuild properties. You don't have to actually run buildalyzer on a project file.

An PR to the full-framework brach seems like the best solution right now, good idea :+1:

Also, thanks for your contribution already!

**Comment by shadow-cs** [comment]
Thanks for the heads-up @Mobrockers @richardwerkman , actually I didn't want to add new project just a MSBuild `props` files that will ensure the stuff I wanted to fix are indeed working. It should not add any more mutations sot it should not increase the pipeline time dramatically (and if so it should only add seconds during initial build, 2s on my machine, the measurement is highly inaccurate though). I will however add this to a different commit so you can decide whether to keep it or not.

**Comment by richardwerkman** [comment]
Closed by #371
